src/tree/updater_gpu_hist.cu

```diff
@@ -324,9 +319,9 @@ class DeviceHistogram {
   }
 
   void Reset() {
-    dh::safe_cuda(cudaMemsetAsync(
-        data_.data().get(), 0,
-        data_.size() * sizeof(typename decltype(data_)::value_type)));
+    auto d_data = data_.data().get();
+      dh::LaunchN(device_id_, data_.size(),
```

Comment:
Is this just a `memset` or `thrust::fill`?

Reply:
Yes, I wanted to avoid cudaMemset due to performance concerns and calls to thrust may soon be synchronous. I might be migrating a lot of thrust code in the near future to other options.